pyaedt/modules/Material.py

```diff
@@ -1992,16 +1993,248 @@ def set_magnetic_coercivity(self, value=0, x=1, y=0, z=0):
+            The default value is ``1mm``.
+        thickness : str
+            Thickness specified in terms of value + unit.
+            The default value is ``0.5mm``.
+        conductivity : float
+            Conductivity. The unit is S/m.
+            The default value is ``0S/m``.
+        coefficient_setup : str, optional
+            Core loss unit.
+            Possible values are: ``w_per_cubic_meter``, ``kw_per_cubic_meter``, ``w_per_lb``, ``w_per_kg``.
+            The default value is ``w_per_cubic_meter``.
+        points_list_at_freq : dict
+            Dictionary where keys are the frequencies (in Hz) and values are lists of points (BP curve).
+            If the core loss model is calculated at one frequency points_list_at_freq has to be provided as a
+            dictionary with one key (single frequency in Hz) and values are lists of points at
+            that specific freq (BP curve).
+
+        Returns
+        -------
+        bool
+            ``True`` if successful or ``False`` if it fails.
+
+        References
+        ----------
+
+        >>> oDefinitionManager.EditMaterial
+
+        Examples
+        --------
+        This example shows how to set a core loss model for a material in case material properties are calculated for
+        core losses at one frequency or core losses versus frequencies (core losses multi curve data).
+        The first case shows how to set properties for core losses at one frequency:
+
+        >>> from pyaedt import Maxwell3d
+        >>> m3d = Maxwell3d()
+        >>> box = m3d.modeler.create_box([-10, -10, 0], [20, 20, 20], "box_to_split")
+        >>> box.material = "magnesium"
+        >>> m3d.materials["magnesium"].set_electrical_steel_coreloss(
+                                                    ... points_list_at_freq={60 : [[0,0], [1,3.5], [2,7.4]]}
+                                                    ... )
+        >>> m3d.release_desktop(True, True)
+
+        The second case shows how to set properties for core losses versus frequencies:
+
+        >>> from pyaedt import Maxwell3d
+        >>> m3d = Maxwell3d()
+        >>> box = m3d.modeler.create_box([-10, -10, 0], [20, 20, 20], "box_to_split")
+        >>> box.material = "magnesium"
+        >>> m3d.materials["magnesium"].set_electrical_steel_coreloss(
+                                                    ... points_list_at_freq={60 : [[0,0], [1,3.5], [2,7.4]],
+                                                    ...                      100 : [[0,0], [1,8], [2,9]],
+                                                    ...                      150 : [[0,0], [1,10], [2,19]]}
+                                                    ... )
+        >>> m3d.release_desktop(True, True)
+
+        """
+        if not isinstance(points_list_at_freq, dict):
+            self.logger.error("Points list at frequencies must be provided as a list of dictionaries.")
+            return False
```

Comment:
Here you should raise a TypeError
```
raise TypeError("Points list at frequency must be provided as a dictionary.")
```
The pyaedt_function_handler will catch the Exception and return False.